### PR TITLE
Add functions to select network interface according to destination IP address

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

**Comment by jukkar** [comment]
Fixed function documentation

**Comment by jukkar** [comment]
recheck

File: include/net/net_if.h [hunk 1]
```diff
@@ -1516,7 +1516,7 @@ static inline void net_if_ipv4_set_gw(struct net_if *iface,
  * will return the default network interface if the best network interface
  * is not found.
  */
-struct net_if *net_if_select_src_iface(struct sockaddr *dst);
+struct net_if *net_if_select_src_iface(const struct sockaddr *dst);
```
Comment (pfalcon) [inline include/net/net_if.h]: Please consider moving this change to the previous commit (which introduced the function declaration), to ensure bisectability ;-).

**Comment by jukkar** [comment]
Fixing the prototype of net_if_select_src_iface() in net_if.h, it was set properly in wrong commit.

**Review by pfalcon — APPROVED** [review]
Makes sense, especially that it's generalization of existing net_if_ipv4_select_src_iface().

**Pull request merged**